supervision/dataset/classification.py

```diff
@@ -0,0 +1,239 @@
+
+        Example:
+            ```python
+            >>> import supervision as sv
+
+            >>> cd = sv.ClassificationDataset(...)
+            >>> train_cd, test_cd = cd.split(split_ratio=0.7, random_state=42, shuffle=True)
+            >>> len(train_cd), len(test_cd)
+            (700, 300)
+            ```
+        """
+        image_names = list(self.images.keys())
+        train_names, test_names = train_test_split(
+            data=image_names,
+            train_ratio=split_ratio,
+            random_state=random_state,
+            shuffle=shuffle,
+        )
+
+        train_dataset = ClassificationDataset(
+            classes=self.classes,
+            images={name: self.images[name] for name in train_names},
+            annotations={name: self.annotations[name] for name in train_names},
+        )
+        test_dataset = ClassificationDataset(
+            classes=self.classes,
+            images={name: self.images[name] for name in test_names},
+            annotations={name: self.annotations[name] for name in test_names},
+        )
+        return train_dataset, test_dataset
+
+    def as_multiclass_folder_structure(
+        self, root_directory_path: str, output_directory_path: str
+    ) -> None:
+        """
+        Saves the dataset as a multi-class folder structure.
+
+        Args:
+            root_directory_path (str): The path to the root directory of the images with which you are working.
+            output_directory_path (str): The path to the directory where the dataset will be saved.
+
+        Example:
+            ```python
+            >>> import supervision as sv
+
+            >>> cd = sv.ClassificationDataset(...)
+
+            >>> cd.as_multiclass_folder_structure(
+            ...     root_directory_path="./images",
+            ...     output_directory_path="./out",
+            ... )
+        """
+        if not os.path.exists(output_directory_path):
+            os.makedirs(output_directory_path)
+
+        for split in ["train", "test", "valid"]:
+            if not os.path.exists(output_directory_path + "/" + split):
+                os.makedirs(output_directory_path + "/" + split)
+
+        train, test = self.split(split_ratio=0.8)
+        train, val = train.split(split_ratio=0.8)
+
+        for split, dataset in zip(["train", "test", "valid"], [train, test, val]):
+            for class_name in self.classes:
+                if not os.path.exists(
+                    output_directory_path + "/" + split + "/" + class_name
+                ):
+                    os.makedirs(output_directory_path + "/" + split + "/" + class_name)
+
+                for image in dataset.images:
+                    if dataset.annotations[image].class_id[0] == self.classes.index(
+                        class_name
+                    ):
+                        full_dir = os.path.join(root_directory_path, image)
+
+                        cv2.imwrite(
+                            output_directory_path
```

Comment:
Please do not build paths using `+ "/" +` . This is not secure. Python has the tools to do that. For example, `os.path.join`.